# Protein turnover from heavy water labeling

The program, d2ome, uses heavy water labeling and LC-MS data to estimate protein degradation rate constants. To use the code first download the executables from the release page of the project. Next, follow the instructions in the "d2ome_Manual.pdf".



## Running d2ome using GUI. 

    d2ome is a stand-alone software. You can start it by downloading and running d2ome_SetUp_GUI.exe. 

The GUI automates filling of the mzML and mzid files from a folder. mzML and mzid file names should match, e.g., SomeFile.mzML, SomeFile.mzid. It is also possible to enter the files manually as in the previous GUI version. In addition, the GUI has an option to load the configuration information from files.txt and quant.state files. The autofill mode starts with “Browse” button. A user can copy and paste a folder path directly into the box. The GUI will sort the files into matching pairs (mzML and mzid). A Tab Controller like the one shown in Figure 1 B should appear. The user will fill the labeling duration (Time) and body water enrichment (BWE) cells for each experiment. Shown in Figure 1 B is an analysis consisting of 18 LC-MS experiments. The labeling time points are: (0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 14, 14, 21, 21). The user will specify the labeling time units (Days or Hours). There are nine labeling timepoints in this example. For each labeling timepoint, there are two biological replicates. The corresponding total body water enrichments are (0, 0.0, 0.0304, 0.0235, 0.0325, 0.0322, 0.0309, 0.0281, 0.0259, 0.0257, 0.0359, 0.0287, 0.0359, 0.0265, etc). 

Alternatively, the user can use the Manual Input button. In this mode, each of the mzML and mzid files and the corresponding labeling duration time and body water enrichment are entered separately. Click on the “Browse” button for the mzML file to find an mzML file. Enter the labeling duration (Time box) and body water enrichment (BWE box), and press the “Add” button to add the data to the list. 

The “Sort” button will sort the input data and order it in a sequence with increasing labeling duration.
Users can use the “Clear All” button to remove all entered files. The “Delete” button will remove selected files only.
“BWE” designates the body water enrichment with D2O that corresponds to the specific labeling experiment. For example, if the body water enrichment is 7%, enter 0.07 into the box under “BWE”. “Rate Constant” method currently allows one options: one-parameter (determine the degradation rate constant). 
Peptide consistency (Figure 1 B) is four. It means that only peptides that have been identified (passed the FDR threshold) and quantified in at least four different timepoints of labeling will be used in the estimation of rate constants for proteins. 
Peptide score (Figure 1 B) is the threshold peptide score (Mascot Ion Score) in peptide-spectrum matches. Currently, the software uses results in the mzid format generated by Mascot. The threshold score is the ion score of Mascot. The program will check that the peptides passed the FDR threshold used in Mascot (using a reversed sequence database).

The mass accuracy to be used in the peak detection.
Enrichment estimation is an option to select between the two monoisotopic RIAs estimation techniques: complete isotope profiles and partial isotope profiles. The monoisotopic RIA is computed from the complete isotope profile of a peptide comprise of up to six mass isotopomers. On the contrary, the partial isotope profiles method utilizes the combination of the ratios from the first three mass isotopomers to improve the number of peptides that have high goodness of fit (R2).
The GUI creates the files.txt and quant.state files (described below), input and parameter files, respectively.
To select the output directory click on the “Browse” button at the bottom of the GUI. One can also copy and paste the folder path to the box next to the button. After the data have been entered, the output directory has been chosen; the quantification is started by the “Start” button, which is next to the Output Directory box, Figure 1 A. 

Note that the software expects that there are NO white spaces in the folder names. 

![Figure 1 A](https://github.com/rgsadygov/d2ome/blob/master/old/images/Picture5.jpg)

Figure 1 A. d2ome GUI before data initialization.


![Figure 1 B](https://github.com/rgsadygov/d2ome/blob/master/old/images/Picture5B.jpg)

Figure 1 B. d2ome GUI in the Load configs mode.



## Running d2ome from command line. 
- Preparing files.txt file 
The location of the mzML and mzID files that were prepared in the first step plus information regarding the labeling duration and the enrichment level should be stored in a text file. The text file is then used when running the software. 
The following is an examples of the text file(called files.txt in our example but can have any name): 
  
          0 B:\Heavy_Water\ A0day_1.mzML B:\Heavy_Water\ A0day_1.mzid 0
          0 B:\Heavy_Water\ A0day_2.mzML B:\Heavy_Water\A0day_2.mzid 0 
          1 B:\Heavy_Water\A1day_1.mzML B:\Heavy_Water\A1day_1.mzid 0.05 
          1 B:\Heavy_Water\\A1day_2.mzML B:\Heavy_Water\A1day_2.mzid 0.05 
          3 B:\Heavy_Water\A3day_1.mzML B:\Heavy_Water\A3day_1.mzid 0.05 
          3 B:\Heavy_Water\A3day_2.mzML B:\Heavy_Water\A3day_2.mzid 0.05 
          5 B:\Heavy_Water\A5day_1.mzML B:\Heavy_Water\A5day_1.mzid 0.05 
          5 B:\Heavy_Water\A5day_2.mzML B:\Heavy_Water\A5day_2.mzid 0.05 
          7 B:\Heavy_Water\A7day_1.mzML B:\Heavy_Water\A7day_1.mzid 0.05 
          7 B:\Heavy_Water\A7day_2.mzML B:\Heavy_Water\A7day_2.mzid 0.05 
          14 B:\Heavy_Water\A14day_1.mzML B:\Heavy_Water\A14day_1.mzid 0.05 
          14 B:\Heavy_Water\A14day_2.mzML B:\Heavy_Water\A14day_2.mzid 0.05 
          21 B:\Heavy_Water\A21day_1.mzML B:\Heavy_Water\A21day_1.mzid 0.05 
          21 B:\Heavy_Water\A21day_2.mzML B:\Heavy_Water\A21day_2.mzid 0.05 
          30 B:\Heavy_Water\A30day_1.mzML B:\Heavy_Water\A30day_1.mzid 0.05 
          30 B:\Heavy_Water\A30day_2.mzML B:\Heavy_Water\A30day_2.mzid 0.05 


In every row, the first number is the number of labeling days, the second element is the address of the mzML file, the third element is the location of the mzid file that corresponds to the preceding mzML file, and the fourth element is the Body Water Enrichment level at that labeling day. For example, in the line, 7 B:\Heavy_Water\A7day_2.mzML B:\Heavy_Water\A7day_2.mzid 0.05 “7” is the number of labeling days for this sample, “B:\Heavy_Water\A7day_2.mzML” is the mzML file, “B:\Heavy_Water\A7day_2.mzid” is the mzid file. “0.05” is the body water enrichment level. Note that two repeating lines with labeling days of “7” indicate the repeats/replicates. There is no limit on the number of replicates in d2ome. In practice, we have used as many as 16 replicates per time point. Computed rate constants are in the reciprocal of the time units specified in this file. 

- Preparing quant.state file 
To overwrite the default parameters of d2ome, one prepares another file: quant.state. Note that in this case the naming is important, and it has to be exactly the same as what is typed here. An example of quant.state file is: 


         mass_accuracy = 15 ppm    // mass accuracy: either in ppm or Da 
          MS1_Type            = 1      // data type of MS1, 1 - centroid, 0 - profile
          protein_score = 50               //minimum protein score 
          peptide_score = 20            // minimum peptide score, ion score in Mascot, default is 30 
          peptide_expectation = 0.05   // maximum peptide expectation in Mascot 
          elutiontimewindow = 1         // time window (mins) to search for elution peak. From the time that highest scoring MS2 was triggered 
          protein_consistency = 4         // minimum number of experiments for protein consistency // default 4 
          peptide_consistency = 4          //mininum number of experiments for a peptide consistency // default 4 
          NParam_RateConst_Fit = 1    // The model for fitting rate constant. Currently, only 1.
          Labeling_time_unit = Days  // Days, Hours



Note also that d2ome is sensitive to what comes before the equal sign and the sign itself. These should not be changed. Also, double slash stands for comment and whatever after that in a line is not read by the software. Therefore, one is only expected to change the numbers.

- Download all binaries into a single folder. Assume, you have downloaded the files into the folder: C:\d2ome_exec. 
- From the directory where you have files.txt and quant.state (and possibly NEH.txt – to specify number of exchangeable hydrogens for each amino acid) files, use the command:

        >  E:\GUI_RUN>C:\d2ome_exec\d2ome.exe files.txt
          
          In the above line, it is assumed that the files.txt and quant.state files are in the E:\GUI_RUN folder.
          The program should start with a message on the output like this one:



![Figure 5 A](https://github.com/rgsadygov/d2ome/blob/master/old/images/Picture6.png)

The results will be in the folder where the program was run. In this case, the folder, E:\GUI_RUN.


## Output

All outputs are reported in csv formatted files. For each protein that passed the thresholds specified in quant.state file, two main output files will be created: ProteinAccession.RateConst.csv and ProteinAccession.Quant.csv. 

The *.Quant.csv file contains comprehensive information about each peptide (which passed the thresholds specified in quant.state file) of a protein. Each peptide entry is a row of information. The information are amino acid sequence, distinctness of the sequence, charge state of the precursor, theoretical m/z ratio of the precursor, theoretical isotope abundances, total labeling (theoretical, before the start of the labeling), precursor m/z value (measured), the highest Mascot Ion score, Mascot expectation, mass accuracy (in ppm), scan number, the integrated abundance of the mass isotopomers (six), elution start and end times that were used to calculate the isotopomer abundances, peak width of the monoisotope in the mass-to-charge domain, total labeling from experimental isotopes. The information is repeated for each experiment. Ion Score of 0 indicates that the peptide was not observed in that particular experiment. The total labeling (molecular percent enrichment) is by default calculated only for the 1st heavy isotope. If the entries are blank for an experiment, it means that the peptide was not fragment in that LC-MS. 

The rows of *.RateConst.csv file of a protein contain: peptide uniqueness (distinct or shared sequence with other peptides) of the sequence, peptide rate constant and corresponding confidence intervals (in unites reciprocal to the time units used in the files.txt file and specified in quant.state), correlation between the fit and experimental data, root mean square error, absolute deviation between the theoretical and experimental isotope profiles (before the start of labeling), peptide charge, sequence mass-to-charge ratio, number of accessible hydrogens (NEH), number of data points (NDP), R2 of the theoretical fit, and averaged abundance of the monoisotope.

Proteins.csv file contains the list of proteins and their Mascot scores for proteins that passed the specified (in quant.state) thresholds.

Analyzed_Proteins.csv file contains a list of all identified proteins, and their corresponding rate constant, confidence interval, standard deviation, abundance, and the number of peptides used to compute protein turnover rate. 

## Visualization

Once the quantification is finished, there will be a message on the screen. Press the “Ok” button. Then press “Visualization” button on the Tab Controller. An output like  the one in Figure 2 should appear on the screen. The program automatically reads the results from the output directory, sorts the proteins by name, and shows the first protein, its rate constant, standard deviation, and its peptides, their characteristics (charge state, R2 of the fit, computed rates, etc.). To look at the results of theoretical fit, one can choose any of the peptides on the screen by “click” or “up”, “down” arrows. The figures for all peptides of all proteins can be saved (in jpeg format) by clicking on the button “Export all proteins”. Alternatively, the protein on the screen can be exported by clicking the “Export Protein_Name” button. The visualization tool can also be used to view previously quantified data. The GUI provides the opportunity to examine the quality of the label incorporation estimation from the data obtained by using the match between-runs (MBR). The user can visualize peptieds identeifeid by MBR by toggling the “Show zero ion score in red” check box. 


![Figure 6 A](https://github.com/rgsadygov/d2ome/blob/master/old/images/Picture10.png)

Figure 2. Output results for protein, 1433E_MOUSE.

## Citation 
The data processing in d2ome was described in[2]. The selection of mass isotopomers for label quantification is described in [1]. 

## References
- Deberneh, Henock M., and Rovshan G. Sadygov. "Software Tool for Visualization and Validation of Protein Turnover Rates Using Heavy Water Metabolic Labeling and LC-MS." International journal of molecular sciences 23.23 (2022): 14620.
- Sadygov RG. Partial Isotope Profiles Are Sufficient for Protein Turnover Analysis Using Closed-Form Equations of Mass Isotopomer Dynamics, Anal Chem 2020;92:14747-14753.
- Sadygov RG, Avva J, Rahman M et al. d2ome, Software for in Vivo Protein Turnover Analysis Using Heavy Water Labeling and LC-MS, Reveals Alterations of Hepatic Proteome Dynamics in a Mouse Model of NAFLD, J Proteome Res 2018;17:3740-3748.
- Sadygov RG. Protein turnover models for LC-MS data of heavy water metabolic labeling, Brief Bioinform 2022.





